### ISSUE TITLE
bitshares login is not useful for sites with profile information (ALMOST ALL SITES)

### ISSUE THREAD

**Issue by sidhujag** [BitShares/bitshares#1349]
Such sites that use this profile information to enhance the user experience will not benefit from bitshares login. Sites such as shopping carts which use profile info to checkout, user can log in but must fill this information out manually every tiem or update their profile manually defeating the purpose of using the login analogous to other social logins such as facebook/twitter.

Suggest that you save profile information under accounts -> gui_data, build ontop of website url to allow more data such as the ones found here: https://github.com/sidhujag/bitshares-drupal7-login/blob/master/sociallogin/socialloginandsocialshare.bitsharesloginapi.inc#L101

**Comment by nathanhourt** [comment]
This is planned, but not yet implemented. We definitely want to have this feature.

**Comment by sidhujag** [comment]
Thanks please keep me in the loop for any design decisions.

Would it be possible to have the web wallet be able to allow login? This would help with the problem of having to open a new browser window on the callback and it would also allow for a more cleaner and flexible interface for a social login

**Comment by nathanhourt** [comment]
I am not aware of a way a website can register itself to handle certain URL schemes, though I imagine browser plugins could do this. It's also very possible that it can be done and I've just never seen it. But if it can't, then the only way to use logins right now is via the Qt wallet.

**Comment by sidhujag** [comment]
Hmm I guess the problem is that the web wallet port and uri isnt known ahead of time.. Google extensions may work but I think bts extension may still be needed and thus wallet handler only

**Comment by nathanhourt** [comment]
Yeah, for the time being, the only way to use logins is to have some BitShares software on your system. And honestly, that makes sense: we're trying to replace identity leasing from companies like Google and Facebook with an identity you yourself own and control, which you can use without needing some other website to vouch for you.

Still, from a convenience perspective, it would be nice, I agree. But it can't (AFAIK) be done.

**Comment by nathanhourt** [comment]
@valzav Can you look at this?

**Comment by nathanhourt** [comment]
As to websites registering themselves as scheme handlers, Troglodactyl just pointed this out:
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/registerProtocolHandler
I'm not sure if it's useful to us or not, given as it only works on the same domain by default, but apparently browsers are thinking about it.

**Comment by sidhujag** [comment]
That would be pretty neat.. we can then use our webwallet to do the authentication.

**Comment by Troglodactyl** [comment]
See this also:  http://updates.html5rocks.com/2011/06/Registering-a-custom-protocol-handler

The drawback here is that for Chrome support you'll have to add a web+ prefix, so either we need to change the primary bts protocol name to "web+bts" so it will work everywhere, or we'll have to deal with some messy compatibility issues.
